pkg/template/jsonpath.go

```diff
@@ -0,0 +1,158 @@
+package template
```

Comment:
hm do you think there's another name we could use besides "template"? in pipelines we are trying to avoid using the term "templating" for the variable substitution we are doing b/c "template" implies additional functionality we want to avoid adding

Reply:
i just realized the type is literally called "trigger template" and this package is probably named after that so just ignore me :woman_facepalming:

Reply:
Given the discussion around JSONPath support in pipeline params, it might be worth having a common jsonpath pkg that both pipeline and triggers could use :thinking: 

(Though that's a yak that we should shave another day :smile: